packages/flutter/lib/src/material/search.dart

```diff
@@ -121,6 +121,7 @@ abstract class SearchDelegate<T> {
     this.searchFieldLabel,
     this.keyboardType,
     this.textInputAction = TextInputAction.search,
+    this.maintainState = false,
```

Comment:
Instead of exposing this in the constructor, maybe just add a getter `maintainState` to the class that returns false. This getter can be overridden by subclasses to do something else. I don't think this needs to be configurable on a per-instance basis.